src/lib/button-toggle/button-toggle.ts

```diff
@@ -461,10 +468,31 @@ export class MdButtonToggle implements OnInit {
     event.stopPropagation();
   }
 
+  _onInputFocus() {
+    // Only show the focus / ripple indicator when the focus was not triggered by a mouse
```

Comment:
maybe you could use the FocusOriginMonitor for this? https://github.com/angular/material2/blob/master/src/lib/core/style/focus-classes.ts